settings/src/script.js

```diff
@@ -83,11 +83,8 @@ function Main( { userId } ) {
 	// The screens where a recent two factor challenge is required.
 	const twoFactorRequiredScreens = [ 'webauthn', 'totp', 'backup-codes' ];
 
-	if ( ! components[ initialScreen ] ) {
-		initialScreen = 'account-status';
-		setScreen( initialScreen );
-		currentUrl.searchParams.set( 'screen', initialScreen );
-		window.history.pushState( {}, '', currentUrl );
+	if ( ! components[ screen ] ) {
+		setScreen( 'account-status' );
```

Comment:
I initially wanted to extract this logic. Just tried out a few methods, there's still one error message remaining. But the original issue has disappeared.